compose_generator.py

```diff
@@ -0,0 +1,276 @@
+            pod_dict["ports"] = [f"{flask_port_counter}:{flask_port_counter}"]
+            flask_port_counter += 1
+
+
+def get_config_file():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-cp', '--config_path', help='Configuration file path',
+                        type=str, default='pipert/utils/config_files/config.yaml')
+    opts, unknown = parser.parse_known_args()
+    config_file = open_config_file(opts.config_path)
+    if isinstance(config_file, str):
+        exit(config_file)
+    return config_file
+
+
+def create_config_file_for_pod(pod_configuration, pod_name):
+    global GENERATED_CONFIG_FILES_FOLDER
+    pod_config_path = GENERATED_CONFIG_FILES_FOLDER + pod_name + ".yaml"
+    with open(pod_config_path, 'w') as pod_config_file:
+        yaml.dump(pod_configuration, pod_config_file)
+        print("Created config file for pod " + pod_name)
+    return pod_config_path
+
+
+def create_pod(pod_template, pod_name, pod_config):
+    # 1. Create the pod config file
+    # 2. Add the pod to the docker compose
+
+    global docker_compose_dictionary
+
+    pod_config_path = create_config_file_for_pod(pod_configuration=pod_config,
+                                                 pod_name=pod_name)
+
+    pod_template["container_name"] = pod_name
+    pod_template["environment"]["CONFIG_PATH"] = pod_config_path
+
+    add_ports_if_needed(pod_template, pod_config["components"])
+    docker_compose_dictionary["services"][pod_name] = pod_template
+
+    return pod_name
+
+
+flask_port_counter = 5000
+
+GENERATED_CONFIG_FILES_FOLDER = "pipert/utils/config_files/"
+
+docker_compose_dictionary = {
+    "version": "3.7",
+    "services": {
+        "redis": {
+            "container_name": "redis",
+            "image": "redis:5.0.7-buster",
+            "logging": {
+                "driver": "none"
+            },
+            "networks": {
+                "default": {
+                    "aliases": [
+                        "redis"
+                    ]
+                }
+            },
+            "restart": "always",
+            "ports": [
+                "6379:6379"
+            ]
+        },
+        "base-pipert": {
+            "container_name": "base-pipert",
+            "logging": {
+                "driver": "none"
+            },
+            "build": {
+                "context": "pipe-base/."
+            },
+            "networks": {
+                "default": {
+                    "aliases": [
+                        "base-pipert"
+                    ]
+                }
+            }
+        }
+    }
+}
+
+config_file = get_config_file()
+
+monitoring_system = config_file["monitoring_system"]
+if monitoring_system.lower() == "prometheus":
+    prometheus_handler(pods=config_file["pods"])
+
+# Create the first pod
+pipeline_first_pod = {
+    "container_name": "",
+    "image": "pipert_pipert",
```

Comment:
Why are you giving the image? We build this image every time?

Reply:
We build the image only the first time on the first pod.